compiler/AST/ModuleSymbol.cpp

```diff
@@ -621,7 +621,10 @@ void ModuleSymbol::addDefaultUses() {
   if (modTag != MOD_INTERNAL) {
     ModuleSymbol* parentModule = toModuleSymbol(this->defPoint->parentSymbol);
 
-    assert (parentModule != NULL);
+    if (parentModule == NULL) {
+      SET_LINENO(this);
+      USR_FATAL("Modules must be declared inside modules or at file-scope");
```

Comment:
I'd probably say "...declared at module- or file-scope" since all code is "inside modules" (i.e., it needs to be just inside the module's scope).  If keeping the current wording, I'd replace "inside" with "within".